pkg/crc/api/handlers.go

```diff
@@ -17,22 +17,32 @@ import (
 	"github.com/code-ready/crc/pkg/crc/version"
 )
 
-func statusHandler(client machine.Client, _ crcConfig.Storage, _ json.RawMessage) string {
+type Handler struct {
+	MachineClient machine.Client
+}
+
+func NewHandler() *Handler {
+	return &Handler{
+		MachineClient: machine.NewClient(),
+	}
+}
+
+func (h *Handler) Status() string {
 	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
-	clusterStatus, _ := client.Status(statusConfig)
+	clusterStatus, _ := h.MachineClient.Status(statusConfig)
 	return encodeStructToJSON(clusterStatus)
 }
 
-func stopHandler(client machine.Client, _ crcConfig.Storage, _ json.RawMessage) string {
+func (h *Handler) Stop() string {
 	stopConfig := machine.StopConfig{
 		Name:  constants.DefaultName,
 		Debug: true,
 	}
-	commandResult, _ := client.Stop(stopConfig)
+	commandResult, _ := h.MachineClient.Stop(stopConfig)
 	return encodeStructToJSON(commandResult)
 }
 
-func startHandler(client machine.Client, crcConfig crcConfig.Storage, args json.RawMessage) string {
+func (h *Handler) Start(crcConfig crcConfig.Storage, args json.RawMessage) string {
```

Comment:
Would it make sense to have the `crcConfig` be part of your handler?

Reply:
not really, we can have the `newConfigFunc` as part of the handler, since we need to re-create the config every time, and only some of the handler functions need it, so thought it'd be better to just pass it those ones who need it.